Detectors/MUON/MID/Simulation/src/Digitizer.cxx

```diff
@@ -0,0 +1,190 @@
+  std::array<double, 2> dist = { area.getYmax() - yPos, yPos - area.getYmin() };
+  addNeighbours(stripIndex, 0, deId, prob, dist, xOffset);
+  return true;
+}
+
+//______________________________________________________________________________
+bool Digitizer::addNeighbours(const Mapping::MpStripIndex& stripIndex, int cathode, int deId, double prob,
+                              const std::array<double, 2>& initialDist, double xOffset)
+{
+  /// Add neighbour strips
+
+  double xOffset2 = xOffset * xOffset;
+  for (int idir = 0; idir < 2; ++idir) {
+    // Search for neighbours in the two directions
+    // up and down for the BP, right and left for the NBP
+    double dist = initialDist[idir];
+    Mapping::MpStripIndex neigh = mMapping.nextStrip(stripIndex, cathode, deId, idir);
+    while (neigh.isValid() && mResponse.isFired(prob, std::sqrt(dist * dist + xOffset2), cathode, deId)) {
+      addStrip(neigh, cathode, deId);
+      dist += mMapping.getStripSize(neigh.strip, cathode, neigh.column, deId);
+      neigh = mMapping.nextStrip(neigh, cathode, deId, idir);
+    }
+  }
+  return true;
+}
+
+//______________________________________________________________________________
+bool Digitizer::hitToDigits(const Hit& hit)
+{
+  /// Generate digits from the hit
+
+  // Convert point from global to local coordinates
+  double xPos = 0.5 * (hit.entrancePoint().x() + hit.exitPoint().x());
+  double yPos = 0.5 * (hit.entrancePoint().y() + hit.exitPoint().y());
+  double zPos = 0.5 * (hit.entrancePoint().z() + hit.exitPoint().z());
+  int deId = hit.GetDetectorID();
+  Point3D<double> localPoint = mTransformer.globalToLocal(deId, xPos, yPos, zPos);
+
+  // First get the touched BP strip
+  Mapping::MpStripIndex stripIndex = mMapping.stripByPosition(localPoint.x(), localPoint.y(), 0, deId);
+
+  // Check if the chamber was efficient
+  bool isEfficientBP, isEfficientNBP;
+  if (!mEfficiencyResponse.isEfficient(deId, stripIndex.column, stripIndex.line, isEfficientBP, isEfficientNBP)) {
+    return false;
+  }
+
+  // Digitize if the RPC was efficient
+  mDigits.clear();
+  double prob = mRandom(mGenerator);
+
+  if (isEfficientBP) {
+    addStrip(stripIndex, 0, deId);
+    MpArea area = mMapping.stripByLocation(stripIndex.strip, 0, stripIndex.line, stripIndex.column, deId);
+    // This is the distance between the hit point and the edges of the strip along y
+    std::array<double, 2> dist = { area.getYmax() - localPoint.y(), localPoint.y() - area.getYmin() };
+    addNeighbours(stripIndex, 0, deId, prob, dist);
+    // Search for neighbours in the close column toward inside
+    addBPStrips(localPoint.x(), localPoint.y(), deId, prob, area.getXmin() - localPoint.x());
+    // Search for neighbours in the close column toward outside
+    addBPStrips(localPoint.x(), localPoint.y(), deId, prob, area.getXmax() - localPoint.x());
+  }
+
+  // Then check the touched NBP strip
```

Comment:
Not sure what you mean here. 
The common parts between BP and NBP are already extracted (see `addStrip` and `addNeighbours`).
The addBPStrips is specific to the BP.
Indeed, for the NBP, the iteration lets me move across columns, so that I can find all of the neighbours in the detection element.
For the BP, the iteration lets me move only inside one column, while I need to call the (more time consuming) `stripByPosition` method if I want to search for the first fired strip in the neighbour columns.
The addBPStrips accounts for this difference.
Since I need to check the column on the left and on the right, I wrote a specific function so that I do not duplicate the code. On the other hand, I pass to the function some parameters calculated elsewhere in order to call the most time consuming functions (`stripByPosition`) only when strictly needed.
I do not see how to modify the code without duplicating the code or reducing the performances.